variables.tf

```diff
@@ -0,0 +1,186 @@
+variable "enable_telemetry" {
+  type        = bool
+  default     = true
+  description = <<DESCRIPTION
+This variable controls whether or not telemetry is enabled for the module.
+For more information see https://aka.ms/avm/telemetryinfo.
+If it is set to false, then no telemetry will be collected.
+DESCRIPTION
+}
+
+# This is required for most resource modules
+variable "resource_group_name" {
+  description = <<DESCRIPTION
+  "The name of the resource group in which to create the Azure Bastion."
+  DESCRIPTION
+  type        = string
+}
+
+variable "virtual_network_name" {
+  description = <<DESCRIPTION
+  "The name of the virtual network where Azure Bastion will be deployed."
+  DESCRIPTION
+  type        = string
+}
+
+
+variable "subnet_name" {
+  description = <<DESCRIPTION
+  "The name of the subnet"
+  DESCRIPTION
+  type        = string
+  default     = "AzureBastionSubnet"
+
+  validation {
+    condition     = var.subnet_name == "AzureBastionSubnet"
+    error_message = "The subnet name must be AzureBastionSubnet."
+  }
+}
+
+
+variable "bastion_host" {
+  description = <<DESCRIPTION
+  "Configuration for Azure Bastion Host"
+  DESCRIPTION
+  type = object({
+    name                = string
+    resource_group_name = string
+    location            = string
+    copy_paste_enabled  = bool
+    file_copy_enabled   = bool
+    sku                 = string
+    ip_configuration = object({
+      name                 = string
+      subnet_id            = string
+      public_ip_address_id = string
+    })
+    ip_connect_enabled     = bool
+    scale_units            = number
+    shareable_link_enabled = bool
+    tunneling_enabled      = bool
+    tags                   = map(string)
+  })
+
+  validation {
+    condition     = basename(var.bastion_host.ip_configuration.subnet_id) == "AzureBastionSubnet"
+    error_message = "The subnet name must be AzureBastionSubnet."
+  }
+  default = {
```

Comment:
Providing a default value for this will make it optional, recommended instead to provide a detailed description which is missing for this variable.